// stardust/dist/views/Item/Item.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _get3 = require('lodash/get');

var _get4 = _interopRequireDefault(_get3);

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _lib = require('../../lib');

var _ItemItems = require('./ItemItems');

var _ItemItems2 = _interopRequireDefault(_ItemItems);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Item = function (_Component) {
  _inherits(Item, _Component);

  function Item() {
    _classCallCheck(this, Item);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Item).apply(this, arguments));
  }

  _createClass(Item, [{
    key: 'render',
    value: function render() {
      var _props = this.props;
      var children = _props.children;
      var className = _props.className;
      var contentClassName = _props.contentClassName;
      var description = _props.description;
      var extra = _props.extra;
      var header = _props.header;
      var image = _props.image;
      var meta = _props.meta;

      var rest = _objectWithoutProperties(_props, ['children', 'className', 'contentClassName', 'description', 'extra', 'header', 'image', 'meta']);

      var _get2 = (0, _get4.default)(image, 'props', {});

      var imageClassName = _get2.className;

      var imageProps = _objectWithoutProperties(_get2, ['className']);

      var classes = (0, _classnames2.default)(className, 'item');
      var imageClasses = (0, _classnames2.default)('ui', imageClassName, 'image');
      var contentClasses = (0, _classnames2.default)(contentClassName, 'content');

      var _description = children || description;

      var content = header || meta || extra ? [header && _react2.default.createElement(
        'div',
        { className: 'header' },
        header
      ), meta && _react2.default.createElement(
        'div',
        { className: 'meta' },
        meta
      ), _description && _react2.default.createElement(
        'div',
        { className: 'description' },
        _description
      ), extra && _react2.default.createElement(
        'div',
        { className: 'extra' },
        extra
      )] : [_description];

      return _react2.default.createElement(
        'div',
        _extends({}, rest, { className: classes }),
        image && _react2.default.createElement(
          'div',
          { className: imageClasses },
          _react2.default.createElement('img', imageProps)
        ),
        '}',
        content && _react2.default.createElement(
          'div',
          { className: contentClasses },
          content
        )
      );
    }
  }]);

  return Item;
}(_react.Component);

Item.propTypes = {
  children: _lib.customPropTypes.every([_lib.customPropTypes.disallow(['description']), _react.PropTypes.node]),
  className: _react.PropTypes.string,
  contentClassName: _react.PropTypes.string,
  description: _lib.customPropTypes.every([_lib.customPropTypes.disallow(['children']), _react.PropTypes.node]),
  extra: _react.PropTypes.node,
  header: _react.PropTypes.node,
  image: _react.PropTypes.node,
  meta: _react.PropTypes.node
};
Item.defaultProps = {
  contentClassName: 'middle aligned'
};
Item._meta = {
  name: 'Item',
  type: _lib.META.TYPES.VIEW
};
Item.Items = _ItemItems2.default;
exports.default = Item;
