// stardust/dist/collections/Breadcrumb/BreadcrumbDivider.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

var _propUtils = require('../../utils/propUtils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * A divider sub-component for Breadcrumb component.
 */
function BreadcrumbDivider(props) {
  var children = props.children;
  var icon = props.icon;
  var className = props.className;

  var classes = (0, _classnames2.default)(className, 'divider');
  var rest = (0, _propUtils.getUnhandledProps)(BreadcrumbDivider, props);

  if (icon) {
    return (0, _propUtils.iconPropRenderer)(icon, _extends({}, rest, { className: classes }));
  }

  return _react2.default.createElement(
    'div',
    _extends({}, rest, { className: classes }),
    children
  );
}

BreadcrumbDivider._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'BreadcrumbDivider',
  type: _Meta2.default.type.collection,
  parent: 'Breadcrumb'
};

BreadcrumbDivider.defaultProps = {
  children: '/'
};

BreadcrumbDivider.propTypes = {
  /** Primary content of the Breadcrumb.Divider. */
  children: _propUtils.customPropTypes.all([_propUtils.customPropTypes.mutuallyExclusive(['icon']), _react.PropTypes.node]),

  /** Classes that will be added to the BreadcrumbDivider className. */
  className: _react.PropTypes.string,

  /** Render as an `Icon` component with `divider` class instead of a `div`. */
  icon: _propUtils.customPropTypes.all([_propUtils.customPropTypes.mutuallyExclusive(['children']), _react.PropTypes.node])
};

exports.default = BreadcrumbDivider;
