// stardust/dist/collections/Breadcrumb/BreadcrumbSection.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

var _propUtils = require('../../utils/propUtils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/**
 * A section sub-component for Breadcrumb component.
 */
function BreadcrumbSection(props) {
  var active = props.active;
  var children = props.children;
  var className = props.className;
  var link = props.link;
  var href = props.href;
  var onClick = props.onClick;

  var classes = (0, _classnames2.default)((0, _propUtils.useKeyOnly)(active, 'active'), className, 'section');
  var rest = (0, _propUtils.getUnhandledProps)(BreadcrumbSection, props);

  var handleClick = function handleClick(e) {
    if (onClick) onClick(e);
  };

  if (link || onClick) {
    return _react2.default.createElement(
      'a',
      _extends({}, rest, { className: classes, onClick: handleClick }),
      children
    );
  }

  if (href) {
    return _react2.default.createElement(
      'a',
      _extends({}, rest, { className: classes, href: href, onClick: handleClick }),
      children
    );
  }

  return _react2.default.createElement(
    'div',
    _extends({}, rest, { className: classes, onClick: handleClick }),
    children
  );
}

BreadcrumbSection._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'BreadcrumbSection',
  type: _Meta2.default.type.collection,
  parent: 'Breadcrumb'
};

BreadcrumbSection.propTypes = {
  /** Style as the currently active section. */
  active: _react.PropTypes.bool,

  /** Primary content of the Breadcrumb.Section. */
  children: _react.PropTypes.node,

  /** Classes that will be added to the BreadcrumbSection className. */
  className: _react.PropTypes.string,

  /** Render as an `a` tag instead of a `div`. */
  link: _propUtils.customPropTypes.all([_propUtils.customPropTypes.mutuallyExclusive(['href']), _react.PropTypes.bool]),

  /** Render as an `a` tag instead of a `div` and adds the href attribute. */
  href: _propUtils.customPropTypes.all([_propUtils.customPropTypes.mutuallyExclusive(['link']), _react.PropTypes.string]),

  /** Render as an `a` tag instead of a `div` and called with event on Section click. */
  onClick: _react.PropTypes.func
};

exports.default = BreadcrumbSection;
