// stardust/dist/elements/Header/_Header.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _lib = require('../../lib');

var _factories = require('../../factories');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _Header(props) {
  var _headerElement = props._headerElement;
  var color = props.color;
  var dividing = props.dividing;
  var block = props.block;
  var attached = props.attached;
  var floated = props.floated;
  var inverted = props.inverted;
  var disabled = props.disabled;
  var textAlign = props.textAlign;
  var icon = props.icon;
  var image = props.image;
  var children = props.children;
  var className = props.className;


  var classes = (0, _classnames2.default)('ui', icon && 'icon', color, (0, _lib.useKeyOnly)(dividing, 'dividing'), (0, _lib.useKeyOnly)(block, 'block'), (0, _lib.useKeyOrValueAndKey)(attached, 'attached'), (0, _lib.useValueAndKey)(floated, 'floated'), (0, _lib.useKeyOnly)(inverted, 'inverted'), (0, _lib.useKeyOnly)(disabled, 'disabled'), (0, _lib.useTextAlignProp)(textAlign), className, 'header');

  var _HeaderComponent = _headerElement;
  var rest = (0, _lib.getUnhandledProps)(_Header, props);

  return _react2.default.createElement(
    _HeaderComponent,
    _extends({ className: classes }, rest),
    (0, _factories.createIcon)(icon),
    (0, _factories.createImage)(image),
    children
  );
}

_Header._meta = {
  name: '_Header',
  type: _lib.META.TYPES.ELEMENT,
  props: {
    attached: ['top', 'bottom'],
    color: _lib.SUI.COLORS,
    floated: _lib.SUI.FLOATS,
    textAlign: _lib.SUI.TEXT_ALIGNMENTS
  }
};

_Header.propTypes = {
  _headerElement: _react.PropTypes.string,
  className: _react.PropTypes.string,
  children: _react.PropTypes.node,

  /** Add an icon by icon name or pass an <Icon /.> */
  icon: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.element]),

  /** Add an image by img src or pass an <Image />. */
  image: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.element]),

  /** Color of the header. */
  color: _react.PropTypes.oneOf(_Header._meta.props.color),

  /** Divide header from the content below it */
  dividing: _react.PropTypes.bool,

  /** Format header to appear inside a content block */
  block: _react.PropTypes.bool,

  /** Attach header  to other content, like a segment */
  attached: _react.PropTypes.oneOf(_Header._meta.props.attached),

  /** Header can sit to the left or right of other content */
  floated: _react.PropTypes.oneOf(_Header._meta.props.floated),

  /** Inverts the color of the header for dark backgrounds */
  inverted: _react.PropTypes.bool,

  /** Show that the header is inactive */
  disabled: _react.PropTypes.bool,

  /** Align header content */
  textAlign: _react.PropTypes.oneOf(_Header._meta.props.textAlign)
};

_Header.defaultProps = {
  _headerElement: 'div'
};

exports.default = _Header;
