// stardust/dist/elements/Header/HeaderSubheader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

var _propUtils = require('../../utils/propUtils');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function HeaderSubheader(props) {
  var children = props.children;
  var className = props.className;


  var classes = (0, _classnames2.default)('sub', className, 'header');

  var rest = (0, _propUtils.getUnhandledProps)(HeaderSubheader, props);

  return _react2.default.createElement(
    'div',
    _extends({ className: classes }, rest),
    children
  );
}

HeaderSubheader._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'HeaderSubheader',
  parent: 'Header',
  type: _Meta2.default.type.element
};

HeaderSubheader.propTypes = {
  /** Primary content of the HeaderSubheader */
  children: _react.PropTypes.node,

  /** Classes to add to the subheader className. */
  className: _react.PropTypes.string
};

exports.default = HeaderSubheader;
