// stardust/dist/addons/Confirm/Confirm.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _bluebird = require('bluebird');

var _bluebird2 = _interopRequireDefault(_bluebird);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Meta = require('../../utils/Meta.js');

var _Meta2 = _interopRequireDefault(_Meta);

var _Modal = require('../../modules/Modal/Modal');

var _Modal2 = _interopRequireDefault(_Modal);

var _ModalContent = require('../../modules/Modal/ModalContent');

var _ModalContent2 = _interopRequireDefault(_ModalContent);

var _ModalFooter = require('../../modules/Modal/ModalFooter');

var _ModalFooter2 = _interopRequireDefault(_ModalFooter);

var _ModalHeader = require('../../modules/Modal/ModalHeader');

var _ModalHeader2 = _interopRequireDefault(_ModalHeader);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Confirm = function (_Component) {
  _inherits(Confirm, _Component);

  function Confirm() {
    var _Object$getPrototypeO;

    var _temp, _this, _ret;

    _classCallCheck(this, Confirm);

    for (var _len = arguments.length, args = Array(_len), _key = 0; _key < _len; _key++) {
      args[_key] = arguments[_key];
    }

    return _ret = (_temp = (_this = _possibleConstructorReturn(this, (_Object$getPrototypeO = Object.getPrototypeOf(Confirm)).call.apply(_Object$getPrototypeO, [this].concat(args))), _this), _this.state = {
      message: 'Are you sure?'
    }, _this.show = function (message) {
      return new _bluebird2.default(function (resolve, reject) {
        _this.setState({ message: message });

        _this.handleAbort = function () {
          return resolve(false);
        };
        _this.handleConfirm = function () {
          return resolve(true);
        };

        _this.refs.modal.showModal();
      }).then(function (isConfirmed) {
        _this.refs.modal.hideModal();
        return isConfirmed;
      });
    }, _temp), _possibleConstructorReturn(_this, _ret);
  }

  _createClass(Confirm, [{
    key: 'render',
    value: function render() {
      var classes = (0, _classnames2.default)(this.props.className);
      return _react2.default.createElement(
        _Modal2.default,
        _extends({}, this.props, { className: classes, ref: 'modal' }),
        _react2.default.createElement(
          _ModalHeader2.default,
          null,
          this.props.header
        ),
        _react2.default.createElement(
          _ModalContent2.default,
          null,
          this.state.message
        ),
        _react2.default.createElement(
          _ModalFooter2.default,
          null,
          _react2.default.createElement(
            'div',
            { className: 'ui button', onClick: this.handleAbort },
            this.props.abortLabel
          ),
          _react2.default.createElement(
            'div',
            { className: 'ui blue button', onClick: this.handleConfirm },
            this.props.confirmLabel
          )
        )
      );
    }
  }]);

  return Confirm;
}(_react.Component);

Confirm.propTypes = {
  abortLabel: _react.PropTypes.string,
  className: _react.PropTypes.string,
  confirmLabel: _react.PropTypes.string,
  header: _react.PropTypes.string
};
Confirm.defaultProps = {
  abortLabel: 'Cancel',
  confirmLabel: 'Yes'
};
Confirm._meta = {
  library: _Meta2.default.library.stardust,
  name: 'Confirm',
  type: _Meta2.default.type.addon
};
exports.default = Confirm;
