// stardust/dist/modules/Progress/Progress.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _without2 = require('lodash/without');

var _without3 = _interopRequireDefault(_without2);

var _round2 = require('lodash/round');

var _round3 = _interopRequireDefault(_round2);

var _clamp2 = require('lodash/clamp');

var _clamp3 = _interopRequireDefault(_clamp2);

var _every2 = require('lodash/every');

var _every3 = _interopRequireDefault(_every2);

var _isUndefined2 = require('lodash/isUndefined');

var _isUndefined3 = _interopRequireDefault(_isUndefined2);

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

var _propUtils = require('../../utils/propUtils');

var _semanticUtils = require('../../utils/semanticUtils');

var sui = _interopRequireWildcard(_semanticUtils);

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } else { var newObj = {}; if (obj != null) { for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) newObj[key] = obj[key]; } } newObj.default = obj; return newObj; } }

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function Progress(props) {
  var active = props.active;
  var attached = props.attached;
  var autoSuccess = props.autoSuccess;
  var color = props.color;
  var children = props.children;
  var className = props.className;
  var disabled = props.disabled;
  var error = props.error;
  var indicating = props.indicating;
  var inverted = props.inverted;
  var label = props.label;
  var percent = props.percent;
  var precision = props.precision;
  var progress = props.progress;
  var size = props.size;
  var success = props.success;
  var total = props.total;
  var value = props.value;
  var warning = props.warning;


  var isAutoSuccess = autoSuccess && (percent >= 100 || value >= total);

  var showProgress = progress || label || !(0, _isUndefined3.default)(precision) || !(0, _every3.default)([total, value], _isUndefined3.default);

  var _percent = void 0;
  if (!(0, _isUndefined3.default)(percent)) {
    _percent = percent;
  } else if (!(0, _isUndefined3.default)(total) && !(0, _isUndefined3.default)(value)) {
    _percent = value / total * 100;
  }

  _percent = (0, _clamp3.default)(_percent, 0, 100);

  if (!(0, _isUndefined3.default)(precision)) {
    _percent = (0, _round3.default)(_percent, precision);
  }

  var progressText = void 0;
  if (label === 'percent' || label === true || (0, _isUndefined3.default)(label)) {
    progressText = _percent + '%';
  } else if (label === 'ratio') {
    progressText = value + '/' + total;
  }

  var classes = (0, _classnames2.default)('ui', size, color, (0, _propUtils.useKeyOnly)(active || indicating, 'active'), (0, _propUtils.useKeyOnly)(isAutoSuccess || success, 'success'), (0, _propUtils.useKeyOnly)(warning, 'warning'), (0, _propUtils.useKeyOnly)(error, 'error'), (0, _propUtils.useKeyOnly)(disabled, 'disabled'), (0, _propUtils.useKeyOnly)(indicating, 'indicating'), (0, _propUtils.useKeyOnly)(inverted, 'inverted'), (0, _propUtils.useValueAndKey)(attached, 'attached'), className, 'progress');

  var rest = (0, _propUtils.getUnhandledProps)(Progress, props);

  return _react2.default.createElement(
    'div',
    _extends({}, rest, { className: classes, 'data-percent': _percent }),
    _react2.default.createElement(
      'div',
      { className: 'bar', style: { width: _percent + '%' } },
      showProgress && _react2.default.createElement(
        'div',
        { className: 'progress' },
        progressText
      )
    ),
    children && _react2.default.createElement(
      'div',
      { className: 'label' },
      children
    )
  );
}

Progress._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'Progress',
  type: _Meta2.default.type.module,
  props: {
    attached: ['top', 'bottom'],
    color: sui.colors,
    label: ['ratio', 'percent'],
    size: (0, _without3.default)(sui.sizes, 'mini', 'huge', 'massive')
  }
};

Progress.propTypes = {
  /** A progress bar can show activity. */
  active: _react.PropTypes.bool,

  /** A progress bar can attach to and show the progress of an element (i.e. Card or Segment). */
  attached: _react.PropTypes.oneOf(Progress._meta.props.attached),

  /** Whether success state should automatically trigger when progress completes. */
  autoSuccess: _react.PropTypes.bool,

  /** A progress bar can have different colors. */
  color: _react.PropTypes.oneOf(Progress._meta.props.color),

  /** Displayed as a label immediately below the progress bar. */
  children: _react.PropTypes.node,

  /** Additional className. */
  className: _react.PropTypes.string,

  /** A progress bar be disabled. */
  disabled: _react.PropTypes.bool,

  /** A progress bar can show a error state. */
  error: _react.PropTypes.bool,

  /** An indicating progress bar visually indicates the current level of progress of a task. */
  indicating: _react.PropTypes.bool,

  /** A progress bar can have its colors inverted. */
  inverted: _react.PropTypes.bool,

  /** Can be set to either to display progress as percent or ratio. */
  label: _propUtils.customPropTypes.all([_propUtils.customPropTypes.any([_propUtils.customPropTypes.require(['percent']), _propUtils.customPropTypes.require(['total', 'value'])]), _react.PropTypes.oneOfType([_react.PropTypes.bool, _react.PropTypes.oneOf(Progress._meta.props.label)])]),

  /** Current percent complete. */
  percent: _propUtils.customPropTypes.all([_propUtils.customPropTypes.mutuallyExclusive(['total', 'value']), _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number])]),

  /** A progress bar can contain a text value indicating current progress. */
  progress: _react.PropTypes.bool,

  /** Decimal point precision for calculated progress. */
  precision: _react.PropTypes.number,

  /** A progress bar can vary in size. */
  size: _react.PropTypes.oneOf(Progress._meta.props.size),

  /** A progress bar can show a success state. */
  success: _react.PropTypes.bool,

  /**
   * For use with value.
   * Together, these will calculate the percent.
   * Mutually excludes percent.
   */
  total: _propUtils.customPropTypes.all([_propUtils.customPropTypes.require(['value']), _propUtils.customPropTypes.mutuallyExclusive(['percent']), _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number])]),

  /**
   * For use with total.
   * Together, these will calculate the percent.
   * Mutually excludes percent.
   */
  value: _propUtils.customPropTypes.all([_propUtils.customPropTypes.require(['total']), _propUtils.customPropTypes.mutuallyExclusive(['percent']), _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.number])]),

  /** A progress bar can show a warning state. */
  warning: _react.PropTypes.bool
};

exports.default = Progress;
