// stardust/dist/modules/Dropdown/DropdownItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Icon = require('../../elements/Icon/Icon');

var _Icon2 = _interopRequireDefault(_Icon);

var _childrenUtils = require('../../utils/childrenUtils');

var _propUtils = require('../../utils/propUtils');

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function DropdownItem(props) {
  var active = props.active;
  var children = props.children;
  var className = props.className;
  var description = props.description;
  var icon = props.icon;
  var onClick = props.onClick;
  var selected = props.selected;
  var text = props.text;
  var value = props.value;

  var rest = _objectWithoutProperties(props, ['active', 'children', 'className', 'description', 'icon', 'onClick', 'selected', 'text', 'value']);

  var handleClick = function handleClick(e) {
    if (onClick) onClick(e, value);
  };

  var classes = (0, _classnames2.default)((0, _propUtils.useKeyOnly)(active, 'active'), (0, _propUtils.useKeyOnly)(selected, 'selected'), 'item', className);
  // add default dropdown icon if item contains another menu
  var iconName = icon || (0, _childrenUtils.someChildType)(children, 'DropdownMenu') && 'dropdown';
  var iconClasses = (0, _classnames2.default)('sd-dropdown-item-icon', iconName, 'icon');

  return _react2.default.createElement(
    'div',
    _extends({}, rest, { className: classes, onClick: handleClick }),
    description && _react2.default.createElement(
      'span',
      { className: 'description' },
      description
    ),
    iconName && _react2.default.createElement(_Icon2.default, { name: iconClasses }),
    text,
    children
  );
}

DropdownItem._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'DropdownItem',
  parent: 'Dropdown',
  type: _Meta2.default.type.module
};

DropdownItem.propTypes = {
  /** Style as the currently chosen item. */
  active: _react.PropTypes.bool,

  /** Primary content. */
  children: _react.PropTypes.node,

  /** Additional className. */
  className: _react.PropTypes.string,

  /** Additional text with less emphasis. */
  description: _react.PropTypes.string,

  /** Add an icon to the item. */
  icon: _react.PropTypes.string,

  /**
   * The item currently selected by keyboard shortcut.
   * This is not the active item.
   */
  selected: _react.PropTypes.bool,

  /** Display text. */
  text: _react.PropTypes.string,

  /** Stored value */
  value: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),

  /** Called on click with (event, value, text). */
  onClick: _react.PropTypes.func
};

exports.default = DropdownItem;
