// stardust/dist/utils/Meta.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _startsWith2 = require('lodash/fp/startsWith');

var _startsWith3 = _interopRequireDefault(_startsWith2);

var _has2 = require('lodash/fp/has');

var _has3 = _interopRequireDefault(_has2);

var _eq2 = require('lodash/fp/eq');

var _eq3 = _interopRequireDefault(_eq2);

var _flow2 = require('lodash/fp/flow');

var _flow3 = _interopRequireDefault(_flow2);

var _curry2 = require('lodash/fp/curry');

var _curry3 = _interopRequireDefault(_curry2);

var _get2 = require('lodash/fp/get');

var _get3 = _interopRequireDefault(_get2);

var _includes2 = require('lodash/fp/includes');

var _includes3 = _interopRequireDefault(_includes2);

var _values2 = require('lodash/fp/values');

var _values3 = _interopRequireDefault(_values2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var LIBRARIES = {
  semanticUI: 'Semantic UI',
  stardust: 'Stardust'
};

var TYPES = {
  addon: 'addon',
  global: 'global',
  collection: 'collection',
  element: 'element',
  view: 'view',
  module: 'module'
};

var LIBRARY_VALUES = (0, _values3.default)(LIBRARIES);
var TYPE_VALUES = (0, _values3.default)(TYPES);

/**
 * Determine if an object qualifies as a META object.
 * It must have the required keys and valid values.
 * @private
 * @param {Object} _meta A proposed Stardust _meta object.
 * @returns {Boolean}
 */
var isMeta = function isMeta(_meta) {
  return (0, _includes3.default)((0, _get3.default)('library', _meta), LIBRARY_VALUES) && (0, _includes3.default)((0, _get3.default)('type', _meta), TYPE_VALUES);
};

/**
 * Extract the Stardust _meta object and optional key.
 * Handles literal _meta objects, classes with _meta, objects with _meta
 * @private
 * @param {function|object} metaArg A class, a component instance, or meta object..
 * @returns {object|string|undefined}
 */
var getMeta = function getMeta(metaArg) {
  // literal
  if (isMeta(metaArg)) return metaArg;

  // from prop
  else if (isMeta((0, _get3.default)('_meta', metaArg))) return metaArg._meta;

    // from class
    else if (isMeta((0, _get3.default)('constructor._meta', metaArg))) return metaArg.constructor._meta;
};

var metaHasKeyValue = (0, _curry3.default)(function (key, val, metaArg) {
  return (0, _flow3.default)(getMeta, (0, _get3.default)(key), (0, _eq3.default)(val))(metaArg);
});
var isType = metaHasKeyValue('type');
var isLibrary = metaHasKeyValue('library');

/**
 * Component meta information.  Used to declaratively classify and identify components.
 * @type {{}}
 */
var META = {
  library: LIBRARIES,
  type: TYPES,

  // library
  isSemanticUI: isLibrary(LIBRARIES.semanticUI),
  isStardust: isLibrary(LIBRARIES.stardust),

  // type
  isType: isType,
  isAddon: isType(TYPES.addon),
  isGlobal: isType(TYPES.global),
  isCollection: isType(TYPES.collection),
  isElement: isType(TYPES.element),
  isView: isType(TYPES.view),
  isModule: isType(TYPES.module),

  // parent
  isParent: (0, _flow3.default)(getMeta, (0, _has3.default)('parent'), (0, _eq3.default)(false)),
  isChild: (0, _flow3.default)(getMeta, (0, _has3.default)('parent')),

  // other
  isPrivate: (0, _flow3.default)(getMeta, (0, _get3.default)('name'), (0, _startsWith3.default)('_'))
};

exports.default = META;
