// stardust/dist/collections/Grid/GridColumn.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _range2 = require('lodash/range');

var _range3 = _interopRequireDefault(_range2);

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _numberToWord = require('../../utils/numberToWord');

var _numberToWord2 = _interopRequireDefault(_numberToWord);

var _Meta = require('../../utils/Meta.js');

var _Meta2 = _interopRequireDefault(_Meta);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var GridColumn = function (_Component) {
  _inherits(GridColumn, _Component);

  function GridColumn() {
    _classCallCheck(this, GridColumn);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(GridColumn).apply(this, arguments));
  }

  _createClass(GridColumn, [{
    key: 'render',
    value: function render() {
      var classes = (0, _classnames2.default)(this.props.className, this.props.width && (0, _numberToWord2.default)(this.props.width) + ' wide', 'column');
      return _react2.default.createElement(
        'div',
        _extends({}, this.props, { className: classes }),
        this.props.children
      );
    }
  }]);

  return GridColumn;
}(_react.Component);

GridColumn.propTypes = {
  children: _react.PropTypes.node,
  className: _react.PropTypes.string,
  style: _react.PropTypes.object,
  width: _react.PropTypes.oneOf((0, _range3.default)(1, 17))
};
GridColumn._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'GridColumn',
  type: _Meta2.default.type.collection,
  parent: 'Grid'
};
exports.default = GridColumn;
