// stardust/dist/deprecatedComponents.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Items = exports.ModalFooter = exports.ModalHeader = exports.ModalContent = exports.ListItem = exports.Buttons = exports.TableColumn = exports.MenuItem = exports.Row = exports.Column = exports.Fields = exports.Field = undefined;

var _lib = require('./lib');

var _Form = require('./collections/Form/Form');

var _Form2 = _interopRequireDefault(_Form);

var _Grid = require('./collections/Grid/Grid');

var _Grid2 = _interopRequireDefault(_Grid);

var _Menu = require('./collections/Menu/Menu');

var _Menu2 = _interopRequireDefault(_Menu);

var _Table = require('./collections/Table/Table');

var _Table2 = _interopRequireDefault(_Table);

var _Buttons2 = require('./elements/Button/Buttons');

var _Buttons3 = _interopRequireDefault(_Buttons2);

var _List = require('./elements/List/List');

var _List2 = _interopRequireDefault(_List);

var _Modal = require('./modules/Modal/Modal');

var _Modal2 = _interopRequireDefault(_Modal);

var _Item = require('./views/Item/Item');

var _Item2 = _interopRequireDefault(_Item);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

/*
 * TODO v1: Remove all deprecated components.
 * These are top level component APIs that have been deprecated.
 * Each top level key includes a message and a replacement component.
 */
var Field = exports.Field = (0, _lib.deprecateComponent)('Field', 'Use "Form.Field" instead.', _Form2.default.Field);

//
// Collections
//
var Fields = exports.Fields = (0, _lib.deprecateComponent)('Fields', 'Use "Form.Fields" instead.', _Form2.default.Fields);

var Column = exports.Column = (0, _lib.deprecateComponent)('Column', 'Use "Grid.Column" instead.', _Grid2.default.Column);
var Row = exports.Row = (0, _lib.deprecateComponent)('Row', 'Use "Grid.Row" instead.', _Grid2.default.Row);

var MenuItem = exports.MenuItem = (0, _lib.deprecateComponent)('MenuItem', 'Use "Menu.Item" instead.', _Menu2.default.Item);

var TableColumn = exports.TableColumn = (0, _lib.deprecateComponent)('TableColumn', 'Use "Table.Column" instead.', _Table2.default.Column);

//
// Elements
//
// TODO deprecate Buttons once updated to v1 API
exports.Buttons = _Buttons3.default;
var ListItem = exports.ListItem = (0, _lib.deprecateComponent)('ListItem', 'Use "List.Item" instead.', _List2.default.Item);

//
// Modules
//
var ModalContent = exports.ModalContent = (0, _lib.deprecateComponent)('ModalContent', 'Use "Modal.Content" instead.', _Modal2.default.Content);
var ModalHeader = exports.ModalHeader = (0, _lib.deprecateComponent)('ModalHeader', 'Use "Modal.Header" instead.', _Modal2.default.Header);
var ModalFooter = exports.ModalFooter = (0, _lib.deprecateComponent)('ModalFooter', 'Use "Modal.Actions" instead.', _Modal2.default.Actions);

//
// Views
//
var Items = exports.Items = (0, _lib.deprecateComponent)('Items', 'Use "Item.Items" instead.', _Item2.default.Items);
