// stardust/dist/index.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Statistic = exports.Items = exports.Item = exports.Dropdown = exports.ModalHeader = exports.ModalFooter = exports.ModalContent = exports.Modal = exports.Progress = exports.Checkbox = exports.Accordion = exports.Rail = exports.Segments = exports.Segment = exports.ListItem = exports.List = exports.Loader = exports.Label = exports.Input = exports.Image = exports.Icon = exports.Header = exports.Flag = exports.Divider = exports.Container = exports.Buttons = exports.Button = exports.TableColumn = exports.Table = exports.Message = exports.MenuItem = exports.Menu = exports.Row = exports.Column = exports.Grid = exports.Fields = exports.Field = exports.Form = exports.Breadcrumb = exports.Textarea = exports.Select = exports.Radio = exports.Confirm = undefined;

require('./utils/jquery');

var _deprecate = require('./utils/deprecate');

var _Confirm2 = require('./addons/Confirm/Confirm');

var _Confirm3 = _interopRequireDefault(_Confirm2);

var _Radio2 = require('./addons/Radio/Radio');

var _Radio3 = _interopRequireDefault(_Radio2);

var _Select2 = require('./addons/Select/Select');

var _Select3 = _interopRequireDefault(_Select2);

var _Textarea2 = require('./addons/Textarea/Textarea');

var _Textarea3 = _interopRequireDefault(_Textarea2);

var _Breadcrumb2 = require('./collections/Breadcrumb/Breadcrumb');

var _Breadcrumb3 = _interopRequireDefault(_Breadcrumb2);

var _Form2 = require('./collections/Form/Form');

var _Form3 = _interopRequireDefault(_Form2);

var _Grid2 = require('./collections/Grid/Grid');

var _Grid3 = _interopRequireDefault(_Grid2);

var _Menu2 = require('./collections/Menu/Menu');

var _Menu3 = _interopRequireDefault(_Menu2);

var _Message2 = require('./collections/Message/Message');

var _Message3 = _interopRequireDefault(_Message2);

var _Table2 = require('./collections/Table/Table');

var _Table3 = _interopRequireDefault(_Table2);

var _Button2 = require('./elements/Button/Button');

var _Button3 = _interopRequireDefault(_Button2);

var _Buttons2 = require('./elements/Button/Buttons');

var _Buttons3 = _interopRequireDefault(_Buttons2);

var _Container2 = require('./elements/Container/Container');

var _Container3 = _interopRequireDefault(_Container2);

var _Divider2 = require('./elements/Divider/Divider');

var _Divider3 = _interopRequireDefault(_Divider2);

var _Flag2 = require('./elements/Flag/Flag');

var _Flag3 = _interopRequireDefault(_Flag2);

var _Header2 = require('./elements/Header/Header');

var _Header3 = _interopRequireDefault(_Header2);

var _Icon2 = require('./elements/Icon/Icon');

var _Icon3 = _interopRequireDefault(_Icon2);

var _Image2 = require('./elements/Image/Image');

var _Image3 = _interopRequireDefault(_Image2);

var _Input2 = require('./elements/Input/Input');

var _Input3 = _interopRequireDefault(_Input2);

var _Label2 = require('./elements/Label/Label');

var _Label3 = _interopRequireDefault(_Label2);

var _Loader2 = require('./elements/Loader/Loader');

var _Loader3 = _interopRequireDefault(_Loader2);

var _List2 = require('./elements/List/List');

var _List3 = _interopRequireDefault(_List2);

var _Segment2 = require('./elements/Segment/Segment');

var _Segment3 = _interopRequireDefault(_Segment2);

var _SegmentSegments = require('./elements/Segment/SegmentSegments');

var _SegmentSegments2 = _interopRequireDefault(_SegmentSegments);

var _Rail2 = require('./elements/Rail/Rail');

var _Rail3 = _interopRequireDefault(_Rail2);

var _Accordion2 = require('./modules/Accordion/Accordion');

var _Accordion3 = _interopRequireDefault(_Accordion2);

var _Checkbox2 = require('./modules/Checkbox/Checkbox');

var _Checkbox3 = _interopRequireDefault(_Checkbox2);

var _Progress2 = require('./modules/Progress/Progress');

var _Progress3 = _interopRequireDefault(_Progress2);

var _Modal2 = require('./modules/Modal/Modal');

var _Modal3 = _interopRequireDefault(_Modal2);

var _Dropdown2 = require('./modules/Dropdown/Dropdown');

var _Dropdown3 = _interopRequireDefault(_Dropdown2);

var _Item2 = require('./views/Item/Item');

var _Item3 = _interopRequireDefault(_Item2);

var _Statistic2 = require('./views/Statistic/Statistic');

var _Statistic3 = _interopRequireDefault(_Statistic2);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

exports.Confirm = _Confirm3.default;

// ----------------------------------------
// Addons
// ----------------------------------------

exports.Radio = _Radio3.default;
exports.Select = _Select3.default;
exports.Textarea = _Textarea3.default;

// ----------------------------------------
// Collections
// ----------------------------------------

exports.Breadcrumb = _Breadcrumb3.default;
exports.Form = _Form3.default;
var Field = exports.Field = (0, _deprecate.deprecateComponent)('Field', 'Use "Form.Field" instead.', _Form3.default.Field);
var Fields = exports.Fields = (0, _deprecate.deprecateComponent)('Fields', 'Use "Form.Fields" instead.', _Form3.default.Fields);

exports.Grid = _Grid3.default;
var Column = exports.Column = (0, _deprecate.deprecateComponent)('Column', 'Use "Grid.Column" instead.', _Grid3.default.Column);
var Row = exports.Row = (0, _deprecate.deprecateComponent)('Row', 'Use "Grid.Row" instead.', _Grid3.default.Row);

exports.Menu = _Menu3.default;
var MenuItem = exports.MenuItem = (0, _deprecate.deprecateComponent)('MenuItem', 'Use "Menu.Item" instead.', _Menu3.default.Item);
exports.Message = _Message3.default;
exports.Table = _Table3.default;
var TableColumn = exports.TableColumn = (0, _deprecate.deprecateComponent)('TableColumn', 'Use "Table.Column" instead.', _Table3.default.Column);

// ----------------------------------------
// Elements
// ----------------------------------------
exports.Button = _Button3.default;
exports.Buttons = _Buttons3.default;
exports.Container = _Container3.default;
exports.Divider = _Divider3.default;
exports.Flag = _Flag3.default;
exports.Header = _Header3.default;
exports.Icon = _Icon3.default;
exports.Image = _Image3.default;
exports.Input = _Input3.default;
exports.Label = _Label3.default;
exports.Loader = _Loader3.default;
exports.List = _List3.default;
var ListItem = exports.ListItem = (0, _deprecate.deprecateComponent)('ListItem', 'Use "List.Item" instead.', _List3.default.Item);

exports.Segment = _Segment3.default;
exports.Segments = _SegmentSegments2.default;
exports.Rail = _Rail3.default;

// ----------------------------------------
// Modules
// ----------------------------------------

exports.Accordion = _Accordion3.default;
exports.Checkbox = _Checkbox3.default;
exports.Progress = _Progress3.default;
exports.Modal = _Modal3.default;
var ModalContent = exports.ModalContent = (0, _deprecate.deprecateComponent)('ModalContent', 'Use "Modal.Content" instead.', _Modal3.default.Content);
var ModalFooter = exports.ModalFooter = (0, _deprecate.deprecateComponent)('ModalFooter', 'Use "Modal.Footer" instead.', _Modal3.default.Footer);
var ModalHeader = exports.ModalHeader = (0, _deprecate.deprecateComponent)('ModalHeader', 'Use "Modal.Header" instead.', _Modal3.default.Header);

exports.Dropdown = _Dropdown3.default;

// ----------------------------------------
// Views
// ----------------------------------------

exports.Item = _Item3.default;
var Items = exports.Items = (0, _deprecate.deprecateComponent)('Items', 'Use "Item.Items" instead.', _Item3.default.Items);
exports.Statistic = _Statistic3.default;
