// stardust/dist/elements/Segment/Segment.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

var _SegmentSegments = require('./SegmentSegments');

var _SegmentSegments2 = _interopRequireDefault(_SegmentSegments);

var _Header = require('../Header/Header');

var _Header2 = _interopRequireDefault(_Header);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

/**
 * A segment is used to create a grouping of related content.
 */

var Segment = function (_Component) {
  _inherits(Segment, _Component);

  function Segment() {
    _classCallCheck(this, Segment);

    return _possibleConstructorReturn(this, Object.getPrototypeOf(Segment).apply(this, arguments));
  }

  _createClass(Segment, [{
    key: 'render',
    value: function render() {
      var heading = _react2.default.createElement(
        _Header2.default.H4,
        null,
        this.props.heading
      );
      var classes = (0, _classnames2.default)('ui', this.props.className, 'segment');

      return _react2.default.createElement(
        'div',
        _extends({}, this.props, { className: classes }),
        this.props.heading && heading,
        this.props.children
      );
    }
  }]);

  return Segment;
}(_react.Component);

Segment.propTypes = {
  /**
   * Segment tag body content.
   */
  children: _react.PropTypes.node,
  className: _react.PropTypes.string,
  /**
   * Optional segment heading.
   */
  heading: _react.PropTypes.any
};
Segment._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'Segment',
  type: _Meta2.default.type.element
};
Segment.Segments = _SegmentSegments2.default;
exports.default = Segment;
