// stardust/dist/modules/Modal/ModalHeader.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _lib = require('../../lib');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function ModalHeader(props) {
  var children = props.children;
  var className = props.className;


  var classes = (0, _classnames2.default)(className, 'header');

  var rest = (0, _lib.getUnhandledProps)(ModalHeader, props);

  return _react2.default.createElement(
    'div',
    _extends({ className: classes }, rest),
    children
  );
}

ModalHeader._meta = {
  name: 'ModalHeader',
  type: _lib.META.TYPES.MODULE,
  parent: 'Modal'
};

ModalHeader.propTypes = {
  /** Primary content of the modal header */
  children: _react.PropTypes.any,

  /** Classes to add to the modal header className */
  className: _react.PropTypes.string
};

exports.default = ModalHeader;
