// stardust/dist/collections/Menu/Menu.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _createClass = function () { function defineProperties(target, props) { for (var i = 0; i < props.length; i++) { var descriptor = props[i]; descriptor.enumerable = descriptor.enumerable || false; descriptor.configurable = true; if ("value" in descriptor) descriptor.writable = true; Object.defineProperty(target, descriptor.key, descriptor); } } return function (Constructor, protoProps, staticProps) { if (protoProps) defineProperties(Constructor.prototype, protoProps); if (staticProps) defineProperties(Constructor, staticProps); return Constructor; }; }();

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _MenuItem = require('./MenuItem');

var _MenuItem2 = _interopRequireDefault(_MenuItem);

var _lib = require('../../lib');

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var Menu = function (_Component) {
  _inherits(Menu, _Component);

  function Menu(props, context) {
    _classCallCheck(this, Menu);

    var _this = _possibleConstructorReturn(this, Object.getPrototypeOf(Menu).call(this, props, context));

    _initialiseProps.call(_this);

    var _this$props = _this.props;
    var activeItem = _this$props.activeItem;
    var children = _this$props.children;

    var firstMenuItem = _lib.childrenUtils.findByType(children, _MenuItem2.default);

    _this.state = {
      activeItem: activeItem || firstMenuItem && firstMenuItem.props.name
    };
    return _this;
  }

  _createClass(Menu, [{
    key: 'render',
    value: function render() {
      var _this2 = this;

      var _props = this.props;
      var activeItem = _props.activeItem;
      var children = _props.children;
      var className = _props.className;

      var rest = _objectWithoutProperties(_props, ['activeItem', 'children', 'className']);

      var classes = (0, _classnames2.default)('ui', className, 'menu');

      var _children = _react.Children.map(children, function (child) {
        var type = child.type;
        var props = child.props;


        return type !== _MenuItem2.default ? child : (0, _react.cloneElement)(child, {
          active: props.name === _this2.state.activeItem || props.name === activeItem,
          __onClick: _this2.handleClickItem
        });
      });
      return _react2.default.createElement(
        'div',
        _extends({}, rest, { className: classes }),
        _children
      );
    }
  }]);

  return Menu;
}(_react.Component);

Menu.propTypes = {
  activeItem: _react.PropTypes.string,
  children: _react.PropTypes.node,
  className: _react.PropTypes.string
};
Menu._meta = {
  name: 'Menu',
  type: _lib.META.TYPES.COLLECTION
};
Menu.Item = _MenuItem2.default;

var _initialiseProps = function _initialiseProps() {
  var _this3 = this;

  this.handleClickItem = function (activeItem) {
    _this3.setState({ activeItem: activeItem });
  };
};

exports.default = Menu;
