// stardust/dist/collections/Menu/MenuItem.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

var _Label = require('../../elements/Label/Label');

var _Label2 = _interopRequireDefault(_Label);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _objectWithoutProperties(obj, keys) { var target = {}; for (var i in obj) { if (keys.indexOf(i) >= 0) continue; if (!Object.prototype.hasOwnProperty.call(obj, i)) continue; target[i] = obj[i]; } return target; }

function MenuItem(_ref) {
  var __onClick = _ref.__onClick;
  var active = _ref.active;
  var children = _ref.children;
  var className = _ref.className;
  var label = _ref.label;
  var name = _ref.name;
  var onClick = _ref.onClick;

  var rest = _objectWithoutProperties(_ref, ['__onClick', 'active', 'children', 'className', 'label', 'name', 'onClick']);

  var handleClick = function handleClick(e) {
    if (__onClick) __onClick(name);
    if (onClick) onClick(name);
  };

  var classes = (0, _classnames2.default)(active && 'active', className, 'item');

  return _react2.default.createElement(
    'a',
    _extends({}, rest, { className: classes, onClick: handleClick }),
    name,
    label && _react2.default.createElement(
      _Label2.default,
      null,
      label
    ),
    children
  );
}

MenuItem._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'MenuItem',
  type: _Meta2.default.type.collection,
  parent: 'Menu'
};

MenuItem.propTypes = {
  __onClick: _react.PropTypes.func,
  active: _react.PropTypes.bool,
  children: _react.PropTypes.node,
  className: _react.PropTypes.string,
  label: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
  name: _react.PropTypes.string,
  onClick: _react.PropTypes.func
};

exports.default = MenuItem;
