// stardust/dist/views/Statistic/StatisticValue.js
'use strict';

Object.defineProperty(exports, "__esModule", {
  value: true
});

var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

var _classnames = require('classnames');

var _classnames2 = _interopRequireDefault(_classnames);

var _react = require('react');

var _react2 = _interopRequireDefault(_react);

var _propUtils = require('../../utils/propUtils');

var _Meta = require('../../utils/Meta');

var _Meta2 = _interopRequireDefault(_Meta);

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

// TODO: This file has disabled shorthand props
// @see https://github.com/TechnologyAdvice/stardust/pull/334

function StatisticValue(props) {
  // const { children, className, content, text } = props
  var children = props.children;
  var className = props.className;
  var text = props.text;

  var classes = (0, _classnames2.default)((0, _propUtils.useKeyOnly)(text, 'text'), className, 'value');
  var rest = (0, _propUtils.getUnhandledProps)(StatisticValue, props);

  // return <div className={classes} {...rest}>{children || content}</div>
  return _react2.default.createElement(
    'div',
    _extends({}, rest, { className: classes }),
    children
  );
}

StatisticValue._meta = {
  library: _Meta2.default.library.semanticUI,
  name: 'StatisticValue',
  parent: 'Statistic',
  type: _Meta2.default.type.view
};

StatisticValue.propTypes = {
  /** Primary content of the StatisticValue. */
  children: _propUtils.customPropTypes.all([_propUtils.customPropTypes.mutuallyExclusive(['content']), _react.PropTypes.node]),

  /** Classes that will be added to the StatisticValue className. */
  className: _react.PropTypes.string,

  // /** Primary content of the StatisticValue. Mutually exclusive with the children prop. */
  // content: customPropTypes.all([
  //   customPropTypes.mutuallyExclusive(['children']),
  //   PropTypes.node,
  // ]),

  /** Format the value with smaller font size to fit nicely beside number values. */
  text: _react.PropTypes.bool
};

exports.default = StatisticValue;
